core/src/broadcast_stage/broadcast_utils.rs

```diff
@@ -26,44 +28,85 @@ pub struct UnfinishedSlotInfo {
     pub parent: Slot,
 }
 
-/// This parameter tunes how many entries are received in one iteration of recv loop
-/// This will prevent broadcast stage from consuming more entries, that could have led
-/// to delays in shredding, and broadcasting shreds to peer validators
-const RECEIVE_ENTRY_COUNT_THRESHOLD: usize = 8;
+#[derive(Default)]
+pub(crate) struct RecvEntriesContext {
+    pending_entry: Option<WorkingBankEntry>,
+}
```

Comment:
I don't think this context is useful.
As long as the channel has any entries we want to drain the channel, and send any entries which are received to downstream for shredding and broadcast.

Buffering these entries for the next time around would only add more overhead which hurts throughput and average latency.

Reply:
This is only retaining the entry which would overflow the 32 shred target byte count. Isn't it more desirable to pack as many entries less than the byte count of 32 shreds rather than overflowing by one entry?

Reply:
Overflowing is fine
In shredder code, if there are more than 32 shreds, the data shreds are split uniformly into erasure batches with at least 32 data shreds in each batch. So if there are 33 data shreds, it will send an erasure batch of 33(data):33(code). If there are 107 data shreds, they are split into 3 batches of 36:36, 36:36 and 35:35 data:code shreds each.

So if it overflows 32 it wouldn't create more coding shreds than data shreds.

Reply:
removed `ctx` and allow overflowing 32 shreds